### PR TITLE
Ensure we don't expose kuebconfig in custom-params.yml

### PR DESCRIPTION
**Pull request by cjeanner**
This patch also prevents loading all of the CI env in the runtime. This
would expose zuul-params.yml content, which contains unfiltered
parameters, among them the kubeconfig one.

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running

### CODE REVIEW

**Comment by cjeanner** [comment]
recheck

**Comment by cjeanner** [comment]
It could make a CI Job reproducer get beyond the nmstate issue faced by the presence of the parameter, toggling the k8s nmstate path.

**Comment by cjeanner** [comment]
/hold

waiting on @cescgina green light.

**Review by marios — APPROVED** [review]
well, it does what the commit message says it does :) lgtm

**Review by raukadah — APPROVED** [review]

**Comment by cjeanner** [comment]
/cancel hold

Let's get it in.

**Pull request merged**
